distribution/lib/Standard/Base/0.0.0-dev/src/Error/Problem_Behavior.enso

```diff
@@ -35,9 +35,10 @@ type Problem_Behavior
        dataflow error. If the value already contained any dataflow error, that
        error takes precedence.
     attach_problem_after : Any -> Any -> Any
-    attach_problem_after self decorated_value ~problem = case self of
-        Ignore ->
-            decorated_value
+    attach_problem_after self decorated_value problem = case self of
+        Ignore -> case decorated_value of
+            _ -> case problem of
+                _ -> decorated_value
```

Comment:
What is the point to this?

Is it for error checking? I think would read a lot clearer with an 
```
if decorated_value.is_error then decorated_value else
    if problem.is_error then problem else decorated_value
```

Reply:
Yes, it is so that any dataflow errors in `decorated_value` take precedence, but if `problem` contained a dataflow error, it is propagated too.

I agree that your proposed syntax reads much better, I will amend this.